osu.Game.Rulesets.Taiko/Difficulty/Skills/Colour.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Rulesets.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Difficulty.Skills;
+using osu.Game.Rulesets.Difficulty.Utils;
+using osu.Game.Rulesets.Taiko.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Difficulty.Skills
+{
+    /// <summary>
+    /// Calculates the colour coefficient of taiko difficulty.
+    /// </summary>
+    public class Colour : Skill
+    {
+        protected override double SkillMultiplier => 1;
+        protected override double StrainDecayBase => 0.4;
+
+        /// <summary>
+        /// Maximum number of entries to keep in <see cref="monoHistory"/>.
+        /// </summary>
+        private const int mono_history_max_length = 5;
+
+        /// <summary>
+        /// Queue with the lengths of the last <see cref="mono_history_max_length"/> most recent mono (single-colour) patterns,
+        /// with the most recent value at the end of the queue.
+        /// </summary>
+        private readonly LimitedCapacityQueue<int> monoHistory = new LimitedCapacityQueue<int>(mono_history_max_length);
+
+        /// <summary>
+        /// The <see cref="HitType"/> of the last object hit before the one being considered.
+        /// </summary>
+        private HitType? previousHitType;
+
+        /// <summary>
+        /// Length of the current mono pattern.
+        /// </summary>
+        private int currentMonoLength = 1;
```

Comment:
I entirely glossed over this. This actually seems to have at least two layers of wrong.

If `StrainValueOf` early-returned if the previous hit type was null, I think this would actually be correct, but it isn't. Also because the first two hit objects don't correspond to difficulty hit objects due to not having two previous objects, the first streak's length can be off up to 2 hits.